From 125fd9e3df5b33b9cd80de1ee3e4fa64c8587387 Mon Sep 17 00:00:00 2001
From: Jan-Marek Glogowski <glogow@fbihome.de>
Date: Wed, 26 Aug 2015 15:29:15 +0000
Subject: tdf#80512 KDE4: fix RTL scrollbar handling

Just invert the scrollbar the same way gtk+ does by checking the
button positions and adjust the drawing style accordingly.

Change-Id: I3475c33e074c1509a73da618a4efa44dc61a1a77
(cherry picked from commit 273df1762707d1d44c9f9b3911e06f7a5a4fa08f)
Reviewed-on: https://gerrit.libreoffice.org/18033
Tested-by: Jenkins <ci@libreoffice.org>
Reviewed-by: Katarina Behrens <Katarina.Behrens@cib.de>

diff --git a/vcl/unx/kde4/KDESalGraphics.cxx b/vcl/unx/kde4/KDESalGraphics.cxx
index 7e1a881..a1a5351 100644
--- a/vcl/unx/kde4/KDESalGraphics.cxx
+++ b/vcl/unx/kde4/KDESalGraphics.cxx
@@ -475,6 +475,8 @@ bool KDESalGraphics::drawNativeControl( ControlType type, ControlPart part,
             option.sliderValue = sbVal->mnCur;
             option.sliderPosition = sbVal->mnCur;
             option.pageStep = sbVal->mnVisibleSize;
+            if (part == PART_DRAW_BACKGROUND_HORZ)
+                option.upsideDown = sbVal->maButton1Rect.Left() > sbVal->maButton2Rect.Left();
 
             //setup the active control...always the slider
             if (sbVal->mnThumbState & ControlState::ROLLOVER)
-- 
cgit v0.10.2

